### PR TITLE
[ENHANCEMENT] TraceQL query editor: add syntax highlighting with Lezer

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
# Description

TraceQL query editor: add syntax highlighting with Lezer

# Screenshots

image

# Checklist

- [X] Pull request has a descriptive title and context useful to a reviewer.
- [X] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [X] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [X] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [X] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [ ] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/tempo-plugin/src/components/TraceQLExtension.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      commentTokens: { line: '//' },
+    },
+  });
+}
+
+export interface CompletionConfig {
+  client?: TempoClient;
+}
+
+export function TraceQLExtension({ client: _client }: CompletionConfig) {
```
Comment (andreasgerstmayr) [inline ui/tempo-plugin/src/components/TraceQLExtension.ts:35]: `client` will be required for auto-completion (in a follow-up PR)

**Review by Nexucis — APPROVED** [review]
so cool !

**Comment by Nexucis** [comment]
maybe we could have the language autocompletion no ?

**Comment by andreasgerstmayr** [comment]
> maybe we could have the language autocompletion no ?

yes, I'm working on that :)

**Comment by Nexucis** [comment]
I was wondering, would you like to become maintainer of Perses @andreasgerstmayr ?

**Comment by andreasgerstmayr** [comment]
> I was wondering, would you like to become maintainer of Perses @andreasgerstmayr ?

that'd be cool, yes!

**Pull request merged**